pkg/api/controller.go

```diff
@@ -559,6 +559,44 @@ func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body apigen.L
 	writeResponse(w, r, http.StatusOK, response)
 }
 
+func (c *Controller) ExternalPrincipalLogin(w http.ResponseWriter, r *http.Request, body apigen.ExternalPrincipalLoginJSONRequestBody) {
+	ctx := r.Context()
+	if c.isExternalPrincipalNotSupported(ctx) {
+		writeError(w, r, http.StatusNotImplemented, "Not implemented")
+		return
+	}
+	c.LogAction(ctx, "external_principal_login", r, "", "", "")
+	externalPrincipal, err := c.Authentication.ExternalPrincipalLogin(ctx, body.IdentityRequest)
+	if c.handleAPIError(ctx, w, r, err) {
```

Comment:
I tested this with lakeFS and fluffy and while fluffy returned 403 Status Code lakeFS returned `500 Internal Server Error`